pkg/auth/auth.go

```diff
@@ -15,21 +16,96 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type HeaderAuthName string
+
+func (h HeaderAuthName) String() string { return string(h) }
+
 // XRegistryAuthHeader is the key to the encoded registry authentication
 // configuration in an http-request header.
-const XRegistryAuthHeader = "X-Registry-Auth"
+const XRegistryAuthHeader HeaderAuthName = "X-Registry-Auth"
+
+// XRegistryConfigHeader is the key to the encoded registry authentication
+// configuration in an http-request header.
+const XRegistryConfigHeader HeaderAuthName = "X-Registry-Config"
```

Comment:
Can you drop a comment that this refers to `build` only?

Reply:
I added "As of API..." messages.